ext/semian/simple_integer.c

```diff
@@ -54,9 +54,12 @@ VALUE semian_simple_integer_initialize(VALUE self, VALUE name)
   TypedData_Get_Struct(self, semian_simple_integer_t, &semian_simple_integer_type, res);
   res->key = generate_key(to_s(name));
 
-  semian_simple_integer_shared_t* data = get_value(self);
+  semian_simple_integer_shared_t* data = get_value(res);
   data->val = 0;
 
+  const int permissions = 0664;
```

Comment:
We should probably move these permissions out somewhere...

Reply:
I've pulled this out into the header for now, looking a bit more closely I wonder if it makes more sense to have this follow whatever permissions are passed from when the resource is created... and also manage the destruction of circuit breakers/sliding windows/simple integers there?